src/Http/Routing/src/RegexInlineRouteConstraintSetup.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Routing.Constraints;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+internal sealed class RegexInlineRouteConstraintSetup : IPostConfigureOptions<RouteOptions>
```

Comment:
Would it be better for this to be `IConfigureOptions` so existing code that might look at `RouteOptions.ConstraintMap` in their own `configure` callback? This won't matter if they call `Configure<RouteOptions>(Action<RouteOptions>)` before `AddRouting()`, but it seems like that would be a little better.